app/services/privacy-module.ts

```diff
@@ -207,4 +221,72 @@ export default class PrivacyModuleService extends Service {
 
     return await this.store.queryRecord('pii-request', {});
   });
+
+  fetchGeoLocationData = task(async (fileId, markSeen = false) => {
+    try {
+      const geoLocation = await this.getGeoLocationRequest.perform(fileId);
+
+      if (
+        geoLocation.status === ENUMS.PM_PII_STATUS.SUCCESS ||
+        geoLocation.status === ENUMS.PM_PII_STATUS.PARTIAL_SUCCESS
+      ) {
+        const queryParams = {
+          geoLocationId: geoLocation.id,
+        };
+
+        const geoLocationDataList = (
+          await this.store.query('geo-location', queryParams)
+        ).toArray() as GeoLocationModel[];
+
+        this.geoLocationDataList = geoLocationDataList;
+        this.geoLocationDataCount = geoLocationDataList.length;
+        this.geoDataAvailable = true;
+
+        if (geoLocation.status === ENUMS.PM_PII_STATUS.PARTIAL_SUCCESS) {
+          this.showCompleteDastScanNote = true;
+        } else {
+          const privacyProjectId =
+            this.router.currentRoute?.parent?.params['app_id'];
```

Comment:
This service could be called in any route. It would be best to supply the `privacyProjectID` as a parameter to this task from the calling component/route. Same thing with PII above.